File: Python/InsertionSort/src/InsertionSort.py
class InsertionSort:

    @staticmethod
    def byAuthor(books):
        booksLength = len(books)
        for index in range(booksLength):
            j = index - 1
            insert = books[index]
            while (j >= 0 and books[j].compareAuthor(insert) > 0):
                books[j+1] = books[j]
                j = j - 1
            books[j+1] = insert

    @staticmethod
    def byNumberPages(books):
        booksLength = len(books)
        for index in range(booksLength):
            j = index - 1
            insert = books[index]
            while (j >= 0 and books[j].compareNumberPages(insert) > 0):
                books[j+1] = books[j]
                j = j - 1
            books[j+1] = insert

    @staticmethod
    def byTitle(books):
        booksLength = len(books)
        for index in range(booksLength):
            j = index - 1
            insert = books[index]
            while (j >= 0 and books[j].compareTitle(insert) > 0):
                books[j+1] = books[j]
                j = j - 1
            books[j+1] = insert

File: C++/InsertionSort/src/InsertionSort.cpp
#include "InsertionSort.h"

void InsertionSort::byAuthor(vector<Book>& _books) {
	for(int i = 1; i < _books.size(); i++) {
		int j = i - 1;
		Book insert = _books[i];
		while(j >= 0 && _books[j].compareAuthor(insert) > 0) {
			_books[j+1] = _books[j];
			j--;
		}
		_books[j+1] = insert;
	}
}

void InsertionSort::byNumberPages(vector<Book>& _books) {
	for(int i = 1; i < _books.size(); i++) {
		int j = i - 1;
		Book insert = _books[i];
		while(j >= 0 && _books[j].compareNumberPages(insert) > 0) {
			_books[j+1] = _books[j];
			j--;
		}
		_books[j+1] = insert;
	}
}

void InsertionSort::byTitle(vector<Book>& _books) {
	for(int i = 1; i < _books.size(); i++) {
		int j = i - 1;
		Book insert = _books[i];
		while(j >= 0 && _books[j].compareTitle(insert) > 0) {
			_books[j+1] = _books[j];
			j--;
		}
		_books[j+1] = insert;
	}
}



